doc/tools/docgram/common.edit_mlg

```diff
@@ -0,0 +1,220 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(* Defines additional productions and edits for use in documentation.  Not compiled into Coq *)
+
+DOC_GRAMMAR
+
+(* additional nts to be spliced *)
+
+LEFTQMARK: [
+| "?"
+]
+
+hyp: [
+| var
+]
```

Comment:
Why this followed by a `SPLICE` instead of a `RENAME`? And what are these BTW?

Reply:
Are you asking about `LEFTQMARK` or `hyp`?

Both these symbols are referenced in the grammar but not defined there, so I need to add productions for them.  `LEFTQMARK` is a token.  `hyp' is defined in OCaml in extraargs.mlg as `register "hyp" wit_var;'  `LEFTQMARK` is spliced because it's better just to show `?` in the grammar.

Reply:
hyp is spliced for simplicity.  Topic for a different PR.